src/BenchmarksApps/HttpClientBenchmarks/Clients/HttpClient/Program.cs

```diff
@@ -0,0 +1,254 @@
+    private static long _exceptionRequests;
+
+    public static async Task<int> Main(string[] args)
+    {
+        var rootCommand = new RootCommand();
+        rootCommand.AddOption(new Option<string>(new string[] { "--url" }, "The server url to request") { Required = true });
+        rootCommand.AddOption(new Option<Version>(new string[] { "--httpVersion" }, "HTTP Version (1.1 or 2.0 or 3.0)") { Required = true });
+        rootCommand.AddOption(new Option<int>(new string[] { "--numberOfClients" }, () => 1, "Number of HttpClients"));
+        rootCommand.AddOption(new Option<int>(new string[] { "--concurrencyPerClient" }, () => 12, "Number of concurrect requests per one HttpClient"));
+        rootCommand.AddOption(new Option<int>(new string[] { "--http11MaxConnectionsPerServer" }, () => 1, "Max number of HTTP/1.1 connections per server"));
+        rootCommand.AddOption(new Option<bool>(new string[] { "--http20EnableMultipleConnections" }, () => false, "Enable multiple HTTP/2.0 connections"));
+        rootCommand.AddOption(new Option<string>(new string[] { "--scenario" }, "Scenario to run") { Required = true });
+        rootCommand.AddOption(new Option<int>(new string[] { "--warmup" }, () => 15, "Duration of the warmup in seconds"));
+        rootCommand.AddOption(new Option<int>(new string[] { "--duration" }, () => 30, "Duration of the test in seconds"));
+
+        rootCommand.Handler = CommandHandler.Create<ClientOptions>(async (options) =>
+        {
+            _options = options;
+            Log("HttpClient benchmark");
+            Log("Options: " + _options);
+
+            Setup();
+            Log("Setup done");
+
+            await Warmup();
+            Log("Warmup done");
+
+            await RunScenario();
+            Log("Scenario done");
+        });
+
+        return await rootCommand.InvokeAsync(args);
+    }
+
+    private static void Setup()
+    {
+        BenchmarksEventSource.Register("IsServerGC", Operations.First, Operations.First, "Server GC enabled", "Server GC is enabled", "");
+        BenchmarksEventSource.Measure("IsServerGC", GCSettings.IsServerGC.ToString());
+
+        BenchmarksEventSource.Register("ProcessorCount", Operations.First, Operations.First, "Processor Count", "Processor Count", "n0");
+        BenchmarksEventSource.Measure("ProcessorCount", Environment.ProcessorCount);
+
+        for (int i = 0; i < _options.NumberOfClients; ++i)
+        {
+            var handler = new SocketsHttpHandler() 
+            {
+                // accept all certs
+                SslOptions = new SslClientAuthenticationOptions { RemoteCertificateValidationCallback = delegate { return true; } },
+                MaxConnectionsPerServer = _options.Http11MaxConnectionsPerServer,
+                EnableMultipleHttp2Connections = _options.Http20EnableMultipleConnections
+            };
+            _httpClients.Add(new HttpClient(handler)
+            {
+                BaseAddress = new Uri(_options.Url!),
```

Comment:
Using the `BaseAddress` and passing a relative Uri to `Get` / `SendAsync` will result in quite a bit of Uri overhead per request.
Unless that was intended, we shouldn't set `BaseAddress`, but create the Uri instance once and re-use it between requests.